src/App.tsx

```diff
@@ -8,55 +8,108 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getPosts } from './api/getPosts';
+import { getUsers } from './api/getUsers';
+import { CommentProvider } from './utils/CommentsContext';
 
 export const App: React.FC = () => {
-  return (
-    <main className="section">
-      <div className="container">
-        <div className="tile is-ancestor">
-          <div className="tile is-parent">
-            <div className="tile is-child box is-success">
-              <div className="block">
-                <UserSelector />
-              </div>
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [users, setUsers] = useState<User[] | null>(null);
+  const [posts, setPosts] = useState<Post[] | null>(null);
```

Comment:
```suggestion
  const [users, setUsers] = useState<User[]>([]);
  const [posts, setPosts] = useState<Post[]>([]);
```
it's better to set an empty array as the default value in such cases. Because then you don't need to check if it's defined before calling some array methods (users.map) or set default value in function parameters or props.
When rendering empty array nothing will be displayed.

From the other side if you really want to check if there is something to render you can check its length

FYI